### PR TITLE
Add practice exercise generator

### PR DESCRIPTION
**Pull request by jiegillet**
Having a generator makes it way, way easier to add exercises, so I thought I would port the one I wrote for the Elm track into Gleam.

Run `bin/generate_practice_exercise.sh <exercise-slug>` and it will create all of the files required for the exercise (minus `tests.toml`, which configlet can do). It works by parsing the JSON of the canonical data, and creating all test cases and solution stub file.

It's not perfect by any means, the biggest advantage is that all the test cases will be in the test file. The data usually requires a lot of regex search-and-replace to get it in a proper shape though. It also tries to guess the types, but that doesn't work that well for non-trivial cases.

I tested the script on a dozen of random exercises, but it's still possible that one exercise could break the script (like a strange variable name in the JSON that is illegal in Gleam or something).

Once concept exercises are being built, it won't be hard to adapt the script to generate all the required files (but it doesn't help nearly as much since there are no test cases).

I'm still a Gleam novice, so the code is probably not perfect. In particular, the JSON parsing game me trouble, I expected to find all the decoders I needed in `gleam_json`, but it only has encoders, so I used `gleam/dynamic` but it doesn't have things like Elm's `succeed`, `lazy` or `andThen`.

### CODE REVIEW

File: bin/generate_practice_exercise.sh [hunk 1]
```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
```
Comment (lpil) [inline bin/generate_practice_exercise.sh:1]: There's a mix of 2 and 4 space indentation in this file. Is that an accident?

File: exercise_generator/gleam.toml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+version = "0.1.0"
+description = "Test and stub file generator for the Exercism Gleam track"
+
+[dependencies]
+gleam_stdlib = "~> 0.26"
+gleam_erlang = "~> 0.17"
+gleam_json = "~> 0.5"
+
+[dev-dependencies]
+gleeunit = "~> 0.9"
```
Comment (lpil) [inline exercise_generator/gleam.toml:11]: Can probably just drop the test code as it is unused.

File: exercise_generator/src/exercise_generator.gleam [hunk 3]
```diff
@@ -0,0 +1,447 @@
...
+
+fn json_data_to_gleam_type(data: JsonData) -> String {
+  case data {
+    JsonNull -> "Result(unknown, Nil)"
+    JsonBool(_) -> "Bool"
+    JsonInt(_) -> "Int"
+    JsonFloat(_) -> "Float"
+    JsonString(_) -> "String"
+    JsonList([]) -> "List(unknown)"
+    JsonList([head, ..]) -> "List (" <> json_data_to_gleam_type(head) <> ")"
```
Comment (lpil) [inline exercise_generator/src/exercise_generator.gleam:66]:
```suggestion
    JsonList([head, ..]) -> "List(" <> json_data_to_gleam_type(head) <> ")"
```

**Review by lpil — APPROVED** [review]
Very nice! This would have saved me a lot of time 😅

File: bin/generate_practice_exercise.sh (hunk 1 above)
Comment (jiegillet) [reply bin/generate_practice_exercise.sh:1]: Indeed, copy/paste mixup.

**Pull request merged**